File: October Challenge/Week_1/C++/Day_4/RemoveCoveredIntervals.cpp
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& inte) {
        
        set<vector<int>> s;
        for(int i=0;i<inte.size();i++){
            int c=inte[i][0];
            int d=inte[i][1];
            bool value=true;
            for(int j=0;j<inte.size();j++){
                if(i==j){
                    continue;
                }else{
                    int a=inte[j][0];
                    int b=inte[j][1];
                    if(c>=a && d<=b){
                        value=false;
                    }
                }
            }
            if(value){
                s.insert(inte[i]);
            }
        }
        return s.size();
        
    }
};

File: October Challenge/Week_1/Java/Day_4/RemoveCoveredIntervals.java
// Problem: https://leetcode.com/problems/remove-covered-intervals/
//Solution:

class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a,b)->(a[0]==b[0])?b[1]-a[1]:a[0]-b[0]);
        int count=0;
        int s=0;
        int f=1;
        int n=intervals.length;
        while(s<n && f<n){
            while(f<n && ((intervals[s][0]<=intervals[f][0]) && (intervals[s][1]>=intervals[f][1]))){
                f++;
                count++;
            }
            s=f;
            f++;
        }
        return n-count;
    }
}